rust/src/file_processor/html_processor.rs

```diff
@@ -0,0 +1,215 @@
+        batch_size: Option<usize>,
+    ) -> Result<Vec<EmbedData>> {
+        let mut embed_data = Vec::new();
+
+        for content in tag_content {
+            let textloader = TextLoader::new(chunk_size);
+            let chunks =
+                match textloader.split_into_chunks(content, SplittingStrategy::Sentence, None) {
+                    Some(chunks) => chunks,
+                    None => continue,
+                };
+
+            if chunks.is_empty() {
+                continue;
+            }
+
+            let tag_type = match tag {
+                "h1" => "header",
+                "h2" => "subheader",
+                "h3" => "subsubheader",
+                "p" => "paragraph",
+                "code" => "code",
+                _ => "paragraph",
+            };
+
+            let metadata = json!({
+                "url": self.origin,
+                "type": tag_type,
+                "full_text": content,
+            });
+
+            let metadata_hashmap: HashMap<String, String> = serde_json::from_value(metadata)?;
+
+            let encodings = embeder.embed(&chunks, batch_size).await?;
+            let embeddings =
+                get_text_metadata(&Rc::new(encodings), &chunks, &Some(metadata_hashmap))?;
+            embed_data.extend(embeddings);
+        }
+
+        Ok(embed_data)
+    }
+}
+
+/// A Struct for processing HTML files.
+pub struct HtmlProcessor;
+
+impl HtmlProcessor {
+    pub fn new() -> Self {
+        Self {}
+    }
+
+    /// Extracts the contents of an HTML file.
+    /// 
+    /// # Arguments
+    /// 
+    /// * `file_path` - The path to the HTML file.
+    /// * `origin` - The original URL of the HTML page, if any. This is required for extracting links.
+    /// 
+    /// # Returns
+    /// 
+    /// Returns a `Result` containing the extracted content as an `HtmlDocument` if successful,
+    /// or an `Error` if an error occurred during any part of the process.
+    pub fn process_html_file(&self, file_path: impl AsRef<std::path::Path>, origin: Option<impl Into<String>>) -> Result<HtmlDocument> {
+        // check if https is in the website. If not, add it.
+        let bytes = std::fs::read(file_path)?;
+        let out = String::from_utf8_lossy(&bytes);
+        self.process_html(out, origin)
+    }
+
+    /// Extracts the contents of an HTML text.
+    ///
+    /// # Arguments
+    ///
+    /// * `html` - The HTML text to be extracted.
+    /// * `origin` - The original URL of the HTML, if any. This is required for extracting links.
+    ///
+    /// # Returns
+    ///
+    /// Returns a `Result` containing the extracted content as an `HtmlDocument` if successful,
+    /// or an `Error` if an error occurred during any part of the process.
+    pub fn process_html(&self, html: impl Into<String>, origin: Option<impl Into<String>>) -> Result<HtmlDocument> {
+        // check if https is in the website. If not, add it.
+        let document = Html::parse_document(&html.into());
```

Comment:
```suggestion
        let origin = origin.map(|o| o.into());
        let document = Html::parse_document(&html.into());
```

Reply:
Sorry about that, I think I've got caching issues with on-the-fly linting :sweat: 
These should both be resolved now